### PR TITLE
[FIX] website: avoid duplicate in _enumerate_pages

### PR DESCRIPTION
**Pull request by JKE-be**
Before this commit, we have redundant loc in sitemap. When we iter the rules from the routing map, the comparison between two same endpoints are not equal since httpocalypse.

After this commit we compare the function of the endpoint to know if we have already processed it or not.

### CODE REVIEW

File: addons/website/controllers/main.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 # Completely arbitrary limits
 MAX_IMAGE_WIDTH, MAX_IMAGE_HEIGHT = IMAGE_LIMITS = (1024, 768)
 LOC_PER_SITEMAP = 45000
-SITEMAP_CACHE_TIME = datetime.timedelta(hours=12)
+SITEMAP_CACHE_TIME = datetime.timedelta(hours=0)
```
Comment (rdeodoo) [inline addons/website/controllers/main.py:34]: Whoops, local dev :p

Comment (JKE-be) [reply addons/website/controllers/main.py]: :angel:

**Comment by Julien00859** [comment]
Hello there, sorry for the late input. How can I generate the sitemap to reproduce the problem? I wonder if we should use `endpoint.func` (=get the non-partialized function) or `endpoint.original_endpoint` (=the original method of the controller, before it was @route decorated).

> There is as many URL duplicate than there is routes defined in the @route() decorator.
/jobs is displayed 8 times because there is 8 routes [defined](https://github.com/odoo/odoo/blame/8a9b418b27dcc9b904413f14cd0443f9a5b10d17/addons/website_hr_recruitment/controllers/main.py#L15-L24) in the method. /jobs/apply/consultant-3 is shown only once because there is only one route [defined](https://github.com/odoo/odoo/blame/8a9b418b27dcc9b904413f14cd0443f9a5b10d17/addons/website_hr_recruitment/controllers/main.py#L90).

I think that's func vs original_endpoint consideration. But I'm not sure about it, I'll test on monday.

**Comment by JKE-be** [comment]
hello

you need to load <domain>/sitemap.xml

and as well delete attachment with name sitemap% to force regeneration as well change the SITEMAP_CACHE_TIME to 0 in addons/website/controllers/main.py to avoid to reuse the attachment version

**Comment by rdeodoo** [comment]
@robodoo rebase-ff

**Comment by JKE-be** [comment]
@robodoo override=ci/style

**Pull request closed without merging**